### DISCUSSION TITLE
A list of the most common Livewire errors

### DISCUSSION THREAD

**Discussion by calebporzio** [livewire/livewire#3327] [5. Announcements]
Over time, a few specific errors have emerged as commonly experienced issues among users.

In this discussion, I'd like to outline the most commonly experienced error messages with the hope of fixing the underlying bugs, or in the cases where it's a userland error, provide better messaging and documentation.

Here are two to start, but please post ones you think are common:
* `Uncaught ReferenceError: $wire is not defined `
* `Uncaught (in promise) TypeError: Cannot read property 'fingerprint' of null`

Also, ideally link to a GitHub gist containing code to reproduce the error (rather than inline code so this thread is less intimidating in length)

**Comment by joshhanley** [comment]
Although the two issues that were just causing this error are fixed, maybe we can consider adding some better console messages around it? It seems to be one of the first things that fires off if something breaks.

![image](https://user-images.githubusercontent.com/882837/125518179-3acaf8e1-8421-44b2-8324-2ae2134e1860.png)

**Comment by danharrin** [comment]
For me, one of the most common is definitely "Converting circular structure to JSON"

#3003, #3075

**Reply by calebporzio** [reply]
Thanks Dan! Any chance for a recreation of the error in a gist?

**Comment by ousid** [comment]
Most of the time get the error `Uncaught (in promise) TypeError: Cannot read property 'fingerprint' of null` it because I don't add the key in element.
```blade
<livewire:name-of-the-component :key="Your Nunique Key" />
```

**Reply by calebporzio** [reply]
Ah, this is a very common one I think, and one I'd like to tackle - can you link to gist replicating the issue? or provide some guidance? Thanks!
